clang-tools-extra/clangd/unittests/CodeCompleteTests.cpp

```diff
@@ -3294,18 +3294,27 @@ TEST(SignatureHelpTest, SkipExplicitObjectParameter) {
     const auto Result = signatureHelp(testPath(TU.Filename), Code.point("c1"),
                                       *Preamble, Inputs, MarkupKind::PlainText);
 
-    EXPECT_EQ(1, Result.signatures.size());
+    EXPECT_EQ(1U, Result.signatures.size());
 
     EXPECT_THAT(Result.signatures[0], AllOf(sig("foo([[int arg]]) -> void")));
   }
   {
     const auto Result = signatureHelp(testPath(TU.Filename), Code.point("c2"),
                                       *Preamble, Inputs, MarkupKind::PlainText);
 
-    EXPECT_EQ(1, Result.signatures.size());
+    EXPECT_EQ(1U, Result.signatures.size());
 
     EXPECT_THAT(Result.signatures[0], AllOf(sig("([[A]], [[int]]) -> void")));
   }
+  {
+    const auto Result = signatureHelp(testPath(TU.Filename), Code.point("c3"),
+                                      *Preamble, Inputs, MarkupKind::PlainText);
+    // TODO: We expect 1 signature here
+    // EXPECT_EQ(1U, Result.signatures.size());
+
+    // EXPECT_THAT(Result.signatures[0], AllOf(sig("([[A]], [[int]]) ->
```

Comment:
Is `(A, int) -> void` really the signature we'd like here?

The compiler seems to accept any type for `self` when using this syntax, e.g. `(&A::foo)(42, 42);` compiles (which is admittedly a bit strange).

So I think the expectation will need to be something like `(auto&&, int) -> void`.